core/src/main/java/hudson/PluginManager.java

```diff
@@ -557,7 +557,7 @@ public void run(Reactor reactor) throws Exception {
             });
 
             // All plugins are loaded. Now we can figure out who depends on who.
-            requires(PLUGINS_PREPARED).attains(COMPLETED).add("Resolving Dependant Plugins Graph", new Executable() {
+            requires(PLUGINS_STARTED, JOB_LOADED).attains(COMPLETED).add("Resolving Dependant Plugins Graph", new Executable() {
```

Comment:
IIUC the real issue here is that there is no hard requirement between PLUGINS_STARTED and EXTENSIONS_AUGMENTED.

I would argue that this should be `requires(PLUGINS_STARTED).attains(EXTENSIONS_AUGMENTED)` and leave `JOB_LOADED` to attain `COMPLETED`